README.md

```diff
@@ -117,6 +118,96 @@ The driver is additionally made available to browsers via CDN:
 
 # Usage
 
+## Creating queries with the `fql` function
+
+
+The `fql` function is your gateway to building safe, reuseable Fauna queries.
+
+It allows you compose queries from sub-queries and variables native to your program. Variables passed in our treated as unexecutable values in Fauna's API - preventing security issues like injection attacks.
+
+for example:
+
+```typescript
+let collectionName = "Pets";
+
+// a reusable sub query to determine a collections existence
+let collectionExists = fql`Collection.byName(${collectionName}) != null`;
+
+client.query(fql`
+  if (${collectionExists}) {
+    "Collection exists!"
+  } else {
+    Collection.create({ name: ${collectionName} })
+    "Collection exists now!"
+  }
+```
+
+This has several advantages:
+
+- You can use `fql` to build a library of subqueries applicable to your domain - and combinable in whatever way you need
+- injection attacks are not possible if you pass input variables into the interpolated (`${i'm interpoloated}`) parts of the query.
+- the driver speaks "pure FQL X" - you can try out some FQL X on the dashboard's terminal and paste it directly into your app like fql``copied from terminal...`` and the query will work as is.
+
+The following subsections show some further examples.
+
+### Pure FQL X via the `fql` function
+The `fql` function can also create pure FQL X queries, for example:
+
+```javascript
+const result = await client.query(fql`
+  let create_user = (params) => if (params.email != null) {
+    User.create(params)
+  } else {
+    null
+  }
+
+  let u = create_user({
+    name: "Alice",
+    email: "alice@site.example",
+  })
+
+  u {
+    id,
+    ts,
+    name,
+    email
+  }
+`);
+```
+
+### Advanced Composition example using `fql` function
+
+
```

Comment:
```suggestion
### Advanced Composition example using `fql` function

```

Reply:
i don't see a difference

Reply:
I don't think that suggestion renders how I tried to write it...

There's an extra line here, that's all.

Reply:
oh one less line

Reply:
will get it next time i'm in here. already merged 😮‍💨

Reply:
Was a tiny nit that doesn't even render :)